### DISCUSSION TITLE
Formula `vdirsyncer` has not implemented #plist, #service or installed a locatable service file

### DISCUSSION THREAD

**Discussion by vaboro** [Homebrew/discussions#3176] [Everyday usage]
Whenever I run

`brew services start vdirsyncer`

I get

`Error: Formula 'vdirsyncer' has not implemented #plist, #service or installed a locatable service file`

<details>
  <summary>This is what's in the Cellar after install</summary>

**server:~ ladmin$ ls -la /usr/local/Cellar/vdirsyncer/0.18.0/
total 88
drwxr-xr-x  11 ladmin  admin    374 12 apr 18:35 .
drwxr-xr-x   3 ladmin  admin    102 12 apr 18:34 ..
drwxr-xr-x   3 ladmin  admin    102 12 apr 18:35 .brew
-rw-r--r--   1 ladmin  staff    490 11 apr 19:02 AUTHORS.rst
-rw-r--r--   1 ladmin  staff  20715 12 apr 18:34 CHANGELOG.rst
-rw-r--r--   1 ladmin  admin   1759 12 apr 18:35 INSTALL_RECEIPT.json
-rw-r--r--   1 ladmin  staff   1594 11 apr 19:02 LICENSE
-rw-r--r--   1 ladmin  staff   2213 12 apr 18:34 README.rst
drwxr-xr-x   3 ladmin  admin    102 12 apr 18:35 bin
drwxr-xr-x   6 ladmin  admin    204 12 apr 18:34 libexec
-rw-r--r--   1 ladmin  staff   1625 12 apr 18:35 vdirsyncer.plist**
</details>

Everything worked fine with vdirsyncer, version 16.8.0 with config file located at ~/.config/vdirsyncer/config
I tried to checkout vdirsyncer, version 16.8.0 from homebrew-core but then during installation I get:

`Error: vdirsyncer: undefined method 'cellar' for #<BottleSpecification:0x00007f868b143318>`

<details>
  <summary>I don't notice anything particularly suspicious in my `brew config` except that I'm on macOS 10.14.6</summary>

**HOMEBREW_VERSION: 3.4.6-60-ge1c1157
ORIGIN: https://github.com/Homebrew/brew
HEAD: e1c1157a769963a4222c1a385bcccf8a9f0c736b
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2fd0231e3f80df5b537cafe214bc7471dbd08a96
Core tap last commit: 15 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.N2Ay6rnqa5/org.xquartz:0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 11.0.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 11.3.1**
</details>

<details>
<summary>`brew doctor` also reveals some unbrewed files</summary>

**Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!**

**Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force**

**If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install**

**Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 11.3.1.**

**Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.**

**Unexpected dylibs:
  /usr/local/lib/libPDFNetC.dylib
  /usr/local/lib/libaspell.15.dylib
  /usr/local/lib/libfragmentzip.0.dylib
  /usr/local/lib/libgeneral.0.dylib
  /usr/local/lib/libimg4tool.0.dylib
  /usr/local/lib/libirecovery.3.dylib
  /usr/local/lib/liboffsetfinder64.0.dylib
  /usr/local/lib/libpspell.15.dylib
  /usr/local/lib/libzip.5.1.dylib**

**Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.**

**Unexpected header files:
  /usr/local/include/aspell.h
  /usr/local/include/libfragmentzip/libfragmentzip.h
  /usr/local/include/libgeneral/macros.h
  /usr/local/include/libirecovery.h
  /usr/local/include/liboffsetfinder64/common.h
  /usr/local/include/pspell/pspell.h
  /usr/local/include/zip.h
  /usr/local/include/zipconf.h**

**Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.**

**Unexpected '.la' files:
  /usr/local/lib/libaspell.la
  /usr/local/lib/libfragmentzip.la
  /usr/local/lib/libgeneral.la
  /usr/local/lib/libimg4tool.la
  /usr/local/lib/libipatcher.la
  /usr/local/lib/libirecovery.la
  /usr/local/lib/liboffsetfinder64.la
  /usr/local/lib/libpspell.la**

**Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.**

**Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/libfragmentzip.pc
  /usr/local/lib/pkgconfig/libgeneral.pc
  /usr/local/lib/pkgconfig/libimg4tool.pc
  /usr/local/lib/pkgconfig/libipatcher.pc
  /usr/local/lib/pkgconfig/libirecovery.pc
  /usr/local/lib/pkgconfig/liboffsetfinder64.pc
  /usr/local/lib/pkgconfig/libzip.pc**

**Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.**

**Unexpected static libraries:
  /usr/local/lib/libcommon.a
  /usr/local/lib/libfragmentzip.a
  /usr/local/lib/libgeneral.a
  /usr/local/lib/libimg4tool.a
  /usr/local/lib/libipatcher.a
  /usr/local/lib/libirecovery.a
  /usr/local/lib/liboffsetfinder64.a
  /usr/local/lib/libxpwn.a**

**Warning: You are using macOS 10.14.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.**
</details>

I posted an issue on vdirsyncer project's issue tracker [page](https://github.com/pimutils/vdirsyncer/issues/977#issue-1202088089) and it seems the issue is not related to vdirsyncer but to Homebrew.

Looking for ways to resolve the issue with vdirsyncer, version 0.18.0.

**Comment by SMillerDev** [comment]
Unless the formula specifies that the plist is at `vdirsyncer.plist` this is not a file that Homebrew knows how to find. Since not all plist files are service definitions we can't just assume that `.plist` means that it's a service file. You make a pull request to make the formula do `mv "vdirsyncer.plist", plist_path` so that brew can find it.

**Reply by vaboro** [reply]
> Unless the formula specifies that the plist is at `vdirsyncer.plist` this is not a file that Homebrew knows how to find. Since not all plist files are service definitions we can't just assume that `.plist` means that it's a service file. You make a pull request to make the formula do `mv "vdirsyncer.plist", plist_path` so that brew can find it.

Thank you! I was able to make the vdirsyncer service work again. It seems an update is needed on vdirsyncer project's side before making a PR to homebrew.

**Reply by ConstantinCezarBegu** [reply]
I am having a similar issue with basically all my services. Is there a way to fix this?

**Reply by ConstantinCezarBegu** [reply]
I have created this issue: https://github.com/Homebrew/homebrew-core/issues/129997
Seems like all services are affected by this which does not make sense to me.
